addon/state-for.js

```diff
@@ -24,8 +24,14 @@ let weakMaps = {};
 * stateFor('state-name').get(this.get('property-name'));
 */
 export default function stateFor(stateName, propertyName) {
+  if (!weakMaps[stateName]) {
```

Comment:
We could restructure it so that in the more common scenario (when the weakmap exists) we only have one object property lookup:

```js
let state = weakMaps[stateName];
if (state === undefined) {
  state = weakMaps[stateName] = new WeakMap();
}
```